### PR TITLE
Force the injected high-priority classes in order to avoid NoClassDefFoundError.

### PR DESCRIPTION
**Pull request by leihuazhe**
Fix when enhance higher priority classes which is put by hashmap causing NoClassDefFoundError

- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by leihuazhe** [comment]
I  change the  skywalking-java-agent  package for our company private using. But, when I launch the agent, here is the phenomenon：
wecom-temp-826069dacbd3a3d8ba1c34b5602ba5ad

The MethodInvocationContext is before MethodInterceptResult, so when load the MethodInvocationContext , will cause  java.lang.NoClassDefFoundError. 

Although the order of high loader classes is not out of order  in the skywalking project.
But I think we should ensure the load sequence should order by the define in the class.

**Comment by leihuazhe** [comment]
image

the MethodInvocationContext is the subclass of MethodInterceptResult.
if classloader load the MethodInvocationContext first, it couldn't find the MethodInterceptResult .so exception below:

java.lang. IllegalStateException Create breakpoint: java.lang.NoClassDefFoundError: com/maple/shine/apm/agent/core/plugin/interceptor/enhance/MethodInterceptResult

this is my customized class.

**Comment by wu-sheng** [comment]
I am not sure this is the case. The class files could be cross-reference. Such as A is using B, and B is using A.

If Byte-buddy is asking for a sorted class map, it should be clear on the method's comment, which I can't find it. Have you checked with byte-buddy author about your case? Because you are using our codebases for private purposes, I can't confirm your change, unless the bytebuddy side could confirm the sequence matters.

**Comment by leihuazhe** [comment]
resonable,I will check the bytebuddy's api, take some expriment.

**Comment by leihuazhe** [comment]
The bytebuddy author's replay:  
_**If there is a dependency, you need to inject the classes in the required order.**_

So maybe we should ensure the sequence when using the ClassInjector.injectRaw

 
https://github.com/raphw/byte-buddy/issues/1299

**Comment by wu-sheng** [comment]
Thanks for the update. You should update changes.md about this fix.

**Comment by leihuazhe** [comment]
finished

File: CHANGES.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ Release Notes.
 * Upgrade gson to 2.8.9.
 * Upgrade netty-codec-http2 to 4.1.79.Final.
 * Fix race condition causing agent to not reconnect after network error
+* Fix enhance high priority classes which is put by hashmap causing NoClassDefFoundError
```
Comment (wu-sheng) [inline CHANGES.md:18]:
```suggestion
* Force the injected high-priority classes in order to avoid NoClassDefFoundError.
```

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**